Integrating weather events into TLO

RachelMurray-Watson opened the pull request:
This PR aims to integrate weather-mediated disruptions into the TLO model. As the disruptions are mediated at the level of a facility, each individual in the model must be "assigned" to facilities of different levels within their region of interest. 

Assumptions: 

- Have the projected disruptions to ANC care; assume this translates to disruptions in other services
- Mode 1 (?)
- Disruption happens at point of healthcare provision, not in healthcare seeking behaviour 
- There's a (weighted) coin toss at the point of processing the HSI event to see if the event goes ahead, based on the proportion of deficit of ANC services from linear model paper
- Appointments are cancelled, not postponed
- Added in logging by region

marghe-molaro commented on src/tlo/methods/healthsystem.py:
```diff
@@ -2254,14 +2282,37 @@ def process_events_mode_0_and_1(self, hold_over: List[HSIEventQueueItem]) -> Non
+            if self.module.services_affected_precip != 'none' and year > 2025:
+                for item in list_of_individual_hsi_event_tuples_due_today_that_have_essential_equipment:
+                    facility_used = self.sim.population.props.at[item.hsi_event.target, 'level_1a']
+                    if facility_used in self.module.parameters['projected_precip_disruptions']['Facility_ID'].values:
```
It might also be clearer to separate the mapping of the modelled and real facilities from the climate considerations, so have dictionaries in the health system ready to be invoked where: 
Real_to_Modelled_Facility_Mapping = {RealFacility_ID : Facility_ID}
and
Modelled_to_Real_Facility_Mapping = {Facility_ID : [RealFacility_ID, RealFacility_ID, ...]}

marghe-molaro commented on src/tlo/methods/healthsystem.py:
```diff
@@ -504,6 +516,16 @@ def __init__(
+        assert climate_model_ensemble_model in ('lowest', 'mean', 'highest')
+        self.climate_model_ensemble_model = climate_model_ensemble_model
+
+        assert services_affected_precip in (None, 'none', 'ANC', 'all')
```
currently only 'none' or 'all' is considered, so remove ANC option for now

marghe-molaro reviewed (commented):
I had a quick look at the changes to the HealthSystem, please let me know if any of my comments are unclear!

RachelMurray-Watson commented on src/tlo/methods/healthsystem.py:
```diff
@@ -2254,14 +2282,37 @@ def process_events_mode_0_and_1(self, hold_over: List[HSIEventQueueItem]) -> Non
+            # And for each indiviudal level event, check to see if there are projected disruptions due to precipitation.
+            if self.module.services_affected_precip != 'none' and year > 2025:
+                for item in list_of_individual_hsi_event_tuples_due_today_that_have_essential_equipment:
+                    facility_used = self.sim.population.props.at[item.hsi_event.target, 'level_1a']
```
Hang-over from previous code; thanks for spotting!

RachelMurray-Watson commented on src/tlo/methods/healthsystem.py:
```diff
@@ -2254,14 +2282,37 @@ def process_events_mode_0_and_1(self, hold_over: List[HSIEventQueueItem]) -> Non
+                            if np.random.binomial(1, prob_disruption) == 1:  # success is delayed appointment
+                                list_of_individual_hsi_event_tuples_due_today_that_have_essential_equipment.remove(
+                                                item)  # Remove item from list? Should I create a seperate list?
+                                self.module.call_and_record_never_ran_hsi_event(hsi_event=item.hsi_event,
```
Per conversation 27th March 2025, decided to "just" cancel all appointments pending further nuancing of healthcare seeking behaviour.

RachelMurray-Watson commented on src/tlo/methods/healthsystem.py:
```diff
@@ -2254,14 +2282,37 @@ def process_events_mode_0_and_1(self, hold_over: List[HSIEventQueueItem]) -> Non
+                                 'service'] == self.module.services_affected_precip),
+                            'disruption'
+                        ]
+                        if len(prob_disruption) > 0:
```
So am I, unsure why I left that there!

RachelMurray-Watson committed:
Fixed high to highest and low to lowest

RachelMurray-Watson committed:
Fixed error of attribute vs parameter

RachelMurray-Watson committed:
Fixed indenting issue that meant that no climate events were being disrupted

RachelMurray-Watson committed:
Shorten for sample run

RachelMurray-Watson committed:
Reverted changes made to test if it's working

RachelMurray-Watson committed:
Reverted changes made to test if it's working

RachelMurray-Watson committed:
This now isn't working to assign parameters... but works in scropt?

RachelMurray-Watson committed:
Get average days

RachelMurray-Watson committed:
Get average days

RachelMurray-Watson committed:
Set up for runs

RachelMurray-Watson committed:
Smaller, faster

RachelMurray-Watson committed:
Check graphs

RachelMurray-Watson committed:
updated to larger population sizes

RachelMurray-Watson committed:
Adding in info for map but don't have health burden by region

RachelMurray-Watson committed:
Initial framework for logging by district

RachelMurray-Watson committed:
Changed to allow for logging by district (and age and time)

RachelMurray-Watson committed:
Changed the column name to district of residence for clarity
Also edited the report_live_years_lost

RachelMurray-Watson committed:
Added line for logging the district of residence in the healthburden and the deaths

RachelMurray-Watson committed:
Extra groupby dimension for dalys stacked by age and time

RachelMurray-Watson committed:
Extra groupby dimension for dalys stacked by age and time

And extra slice for indexing at end of year

RachelMurray-Watson committed:
Changed so clinics were assigned as categories

RachelMurray-Watson committed:
renamed facility level -> facility level type for clarity

RachelMurray-Watson committed:
removed old function based on population weights

RachelMurray-Watson committed:
edited comment to clarify logging by district

RachelMurray-Watson committed:
Added test for logging by district

RachelMurray-Watson committed:
Initial adding in of rescheduling HSI events.

- As a placeholder, if the disruption probability is high, cancelled.

- If not, and there is perfect healthcare seeking, delay is 1 month (to be varied).

- Delay is also 1 month for imperfect healthcare seeking, hopefully to be nuanced

RachelMurray-Watson committed:
could make at level of hsi?

RachelMurray-Watson committed:
erroneous division

RachelMurray-Watson committed:
REmoved unnecessary prints

RachelMurray-Watson committed:
Test isn't working, again won't read in parameters

RachelMurray-Watson committed:
Corrected issue of not using the healthsystem module to access _add_hsi_event_queue_item_to_hsi_event_queue.

also, included modification in healtchare seeking behaviour

RachelMurray-Watson committed:
Reverted changes to hsi_event.py

RachelMurray-Watson committed:
Formatting

RachelMurray-Watson committed:
Sample file for test

RachelMurray-Watson committed:
Sample file for test

RachelMurray-Watson committed:
Error in assignation of child healthcare seeking

RachelMurray-Watson committed:
added more scenarios

marghe-molaro commented on src/tlo/methods/healthsystem.py:
```diff
@@ -2254,17 +2284,84 @@ def process_events_mode_0_and_1(self, hold_over: List[HSIEventQueueItem]) -> Non
+                                climate_disrupted = True
+                                response_to_disruption = 'delay'
+                                if response_to_disruption == 'delay':
+                                    if self.sim.modules['HealthSeekingBehaviour'].force_any_symptom_to_lead_to_healthcareseeking:
```
To discuss over meeting: should force_any_symptom_to_lead_to_healthcareseeking to triumph over response_to_disruption == 'cancel' too?

In other words, should response_to_disruption = 'cancel' be incompatible with force_any_symptom_to_lead_to_healthcareseeking

marghe-molaro reviewed (commented):
This is looking great. I think my main higher-level comment is that it would be good to move some of this into functions + reorganise some of the if loops, so that they are clearer and more efficient. 

Small point for discussion: should always_seek_care and response_to_disruption == ‘cancel’ be incompatible/which should win?

I propose three functions:
In your (I think for now hypothetical?) climate module:
check_if_hsi_experiences_climate_disruption(): current lines 2296 to 2313 

In the HealthSeekingBehaviour:
will_seek_care_again_after_disruption(): # current lines 2323 - 2342

And in HSI_Event.py
reschedule(): call did not run first and then move by given delay


Then all we would need is:

`climate_disruption = check_if_hsi_experiences_climate_disruption()`

`if not climate_disruption:` # Ensuring that lack of climate disrupt is necessary condition for equipment check
`	If equipment_available:`
`		Add to list of hsis to go ahead `
`	else:`
`		Call never_ran()`

`elif climate_disruption:`
`	If response_to_disruption == ‘cancel’:`
`		Call never_ran()`
`	elif response_to_disruption == ‘delay’: `
`		if always healthcare seek`
`			reschedule() `
`		Else:`
`			will_seek_again = will_seek_care_again_after_disruption() `
`			If will_seek_again = `	
`				reschedule()`
`			else:`
`				Call never_ran()`
`

RachelMurray-Watson committed:
Edited comment

RachelMurray-Watson committed:
Changed fac_id to fac_level for clarity (in climate disruptions)

RachelMurray-Watson committed:
Changed year to first check in climate disruption as most frequent check to occur

RachelMurray-Watson committed:
Changed the ordering based on Margherita's suggestion. Now, checks climate disruptions, then equipment

RachelMurray-Watson commented on src/tlo/methods/healthsystem.py:
```diff
@@ -2254,17 +2284,84 @@ def process_events_mode_0_and_1(self, hold_over: List[HSIEventQueueItem]) -> Non
+                            prob_disruption = float(prob_disruption.iloc[0]) # to get average days
+                            if np.random.binomial(1, prob_disruption) == 1:  # success is delayed appointment
+                                climate_disrupted = True
+                                response_to_disruption = 'delay'
```
For this, does it matter that the variable is rewritten every time the calculation is made? i.e., if it is the healthcare seeking behaviour that determines cancellation or otherwise, how would it work to have a general "response_to_disruption" declared?

RachelMurray-Watson committed:
Added number_weeks in delay in seeking healthcare after delay due to weather. Also added nature of disruption as a parameter

RachelMurray-Watson committed:
REmoved reference to disruption as parameter

RachelMurray-Watson committed:
Actually formalised delay_in_seeking_care_weather parameter

RachelMurray-Watson committed:
Actually formalised delay_in_seeking_care_weather parameter

RachelMurray-Watson committed:
Added delay_in_seeking_care_weather to parameter file

RachelMurray-Watson committed:
Formatting

RachelMurray-Watson committed:
Added in delay as a parameter

RachelMurray-Watson committed:
Initial file for module on climate disruptions

RachelMurray-Watson committed:
created function testing whether disruption happens

RachelMurray-Watson committed:
typo

RachelMurray-Watson committed:
Allowed to take just one entry in the dataframe

RachelMurray-Watson committed:
Broke it

RachelMurray-Watson committed:
minor

RachelMurray-Watson committed:
Fixed age -> age_years (have hashing outs i dont need)

RachelMurray-Watson committed:
Fixed typo

RachelMurray-Watson committed:
Fixed so that it passes through a single individual

RachelMurray-Watson committed:
Sample run

RachelMurray-Watson committed:
Added higher pop for runs

RachelMurray-Watson committed:
To unpickle log files

RachelMurray-Watson committed:
fewer years

RachelMurray-Watson committed:
Removed the edits for testing

RachelMurray-Watson committed:
Typo

RachelMurray-Watson committed:
Fixed the year check

RachelMurray-Watson committed:
new directoty

RachelMurray-Watson committed:
Add a dictionary of the population size in each district each year

RachelMurray-Watson committed:
test

RachelMurray-Watson committed:
Made so that each district is plotted

Need to edit as too messy - do not actually need info on each district's dalys, could just do total

RachelMurray-Watson committed:
Edited to allow for breakdown by district. Should add a map

RachelMurray-Watson committed:
Test

RachelMurray-Watson committed:
Have bar plots of the change in dalys by distirct over time

RachelMurray-Watson committed:
Added map of Malawi

RachelMurray-Watson committed:
Added functions for weather-delayed and cancelled HSI events

BUT NOT INVOKED

RachelMurray-Watson committed:
Have invoked, need to test

RachelMurray-Watson committed:
Added climate disruptions to summary logger

RachelMurray-Watson committed:
Added requisite storage

RachelMurray-Watson committed:
Typo

RachelMurray-Watson committed:
Tidy

RachelMurray-Watson committed:
Tidy

RachelMurray-Watson committed:
Tidy

RachelMurray-Watson committed:
Script for plotting cancelled and delayed appointments (due to weather)

RachelMurray-Watson committed:
Tidy

RachelMurray-Watson committed:
Added a graph of total, as doesn't make sense to have sum over a rate/

RachelMurray-Watson committed:
Added relative to baseline

RachelMurray-Watson committed:
Set up for runs

RachelMurray-Watson committed:
increase pop size

RachelMurray-Watson committed:
Typo

RachelMurray-Watson committed:
Retrieve the temperature data from era4

RachelMurray-Watson committed:
Fixed indexing issue

RachelMurray-Watson committed:
Climate-cancelled and delayed HSIs

RachelMurray-Watson committed:
Fixed typo - presumed disruption was a percentage, but it was a raw number.

RachelMurray-Watson committed:
Updated directory

Added script

RachelMurray-Watson committed:
Added new graphs of the weather-disrupted

RachelMurray-Watson committed:
Updated date

RachelMurray-Watson committed:
Merge branch 'refs/heads/master' into rmw/climate_change_switch

# Conflicts:
#	docs/publications.bib
#	resources/healthsystem/consumables/README.md

RachelMurray-Watson committed:
Potential typo

RachelMurray-Watson committed:
Assumed percentage when was actually whole number

RachelMurray-Watson committed:
Merge remote-tracking branch 'origin/rmw/climate_change_switch' into rmw/climate_change_switch

RachelMurray-Watson committed:
Delete src/scripts/climate_change_heat/calculating_relative_humidity.ipynb

didn't need here

RachelMurray-Watson committed:
Formatting

RachelMurray-Watson committed:
Delete resources/climate_change_impacts/ResourceFile_Precipitation_Disruptions.csv

Unnecessary general file

RachelMurray-Watson committed:
Changed date

RachelMurray-Watson committed:
Created new function for when there's a dictionary of counts

RachelMurray-Watson committed:
Merge remote-tracking branch 'origin/rmw/climate_change_switch' into rmw/climate_change_switch

RachelMurray-Watson committed:
changed year, added in series generator that accounts for data structure of hsi count logs

RachelMurray-Watson committed:
starts in 2025

RachelMurray-Watson committed:
Changed so that the maps show difference in DALYS from baseline

RachelMurray-Watson committed:
Added notebook to git ignore

RachelMurray-Watson committed:
Added notebook to git ignore

RachelMurray-Watson committed:
Year edits

RachelMurray-Watson committed:
Final aesthetics to skip baseline

RachelMurray-Watson committed:
Year edits

RachelMurray-Watson committed:
Year edits

RachelMurray-Watson committed:
Altered so that there is a random number generator in the healthcare seeking behaviour model. This will then determine whether healthcare seeking goes ahead as a binary, not as a probability.

RachelMurray-Watson committed:
Changed year

RachelMurray-Watson committed:
Test for climate logging

Setting the parameter isn't working

RachelMurray-Watson committed:
Updated so that parameters change

RachelMurray-Watson committed:
Typo

RachelMurray-Watson committed:
Typo

RachelMurray-Watson committed:
Json update

RachelMurray-Watson committed:
Changed so not maximal healthcare seeking

RachelMurray-Watson committed:
Aesthetics

RachelMurray-Watson committed:
Double recorded never ran hsi

RachelMurray-Watson committed:
Consistency in graphs

RachelMurray-Watson committed:
Restored original probability

RachelMurray-Watson committed:
who knows

RachelMurray-Watson committed:
Shorten

RachelMurray-Watson committed:
District sums stopped working...

RachelMurray-Watson committed:
Restored original figures

RachelMurray-Watson committed:
Plotting fine

RachelMurray-Watson committed:
Introduced:

1) Parameters to scale the delay in reseeking healthcare based on if the HSI was urgent
2) Parameter to rescale the probability of disruption
3) Not implement but included reading in, etc, of the rescaling for probability of reseeking healtchare

RachelMurray-Watson committed:
Parsimony: one scale factor for all priority policies

RachelMurray-Watson committed:
Added in min(prob, 1) to the healthcareseeking
Also changed to 2010, guaranteed disruption etc., for test

RachelMurray-Watson committed:
Correct generation of random number

RachelMurray-Watson committed:
Restored actual checks etc for scenario file

RachelMurray-Watson committed:
Changed scenario file for the following:
added a parameter grid randomly generate parameter sets, varying the scale_factor_delay_in_seeking_care_weather, rescaling_prob_seeking_after_disruption, and rescaling_prob_disruption.

Overwrote the tclose to 200 days

Removed extra scenarios

RachelMurray-Watson committed:
Changed so that the climate-disrupted events also have the facility ID saved to the main logger

RachelMurray-Watson committed:
Added notes to match overleaf text

RachelMurray-Watson committed:
Update date

RachelMurray-Watson committed:
Changed to regular intervals

RachelMurray-Watson committed:
Now includes scale_factor_severity_disruption_and_delay, which divides the p(disruption) (an indicator of incident severity) to then modify the delay in reseeking healthcare.

RachelMurray-Watson committed:
Updated parameter values for search.

1) Increased scale_factor_delay_in_seeking_care_weather to account for many-month delays (this is at most multiplied by 2 in the naive priority policy)

2) Reduce range of interest in reseeking healthcare (half as likely, as likely, 1.5 times as likely) to incorportate the randomness from the draw anyways

3) Included scale_factor_severity_disruption_and_delay, with max extension of x10

RachelMurray-Watson committed:
Improved logging message

RachelMurray-Watson committed:
Added new param scale_factor_severity_disruption_and_delay

RachelMurray-Watson committed:
Changed never_ran and all logger to be the summary logger (to get facility-level info)

RachelMurray-Watson committed:
Facility-level trajectories for HSIs

RachelMurray-Watson committed:
jsopn

RachelMurray-Watson committed:
get summary stats

RachelMurray-Watson committed:
increased t close

made pop smaller so can test/visualise the facility-level trajectories

RachelMurray-Watson committed:
typo

RachelMurray-Watson committed:
typo

RachelMurray-Watson committed:
Fixed so that it's the max of the priority days that is multiplied by the scale

RachelMurray-Watson committed:
Updated for submission

RachelMurray-Watson committed:
Updated for submission

RachelMurray-Watson committed:
included baseline scenario

RachelMurray-Watson committed:
changed json

RachelMurray-Watson committed:
Fixed so that it doesn't log every HSI unless it's one of [4-11], which are in Blantyre

RachelMurray-Watson committed:
Saving % disrupted

RachelMurray-Watson committed:
Added properties graph to see dependencies

RachelMurray-Watson committed:
Updated with directory and means of reading files when one large log file

RachelMurray-Watson committed:
Create smaller log files from the one large log file

RachelMurray-Watson committed:
Graphing

RachelMurray-Watson committed:
Try new param grid

RachelMurray-Watson committed:
Change print

RachelMurray-Watson committed:
Reset to correct parameters. It takes random draw from parameter grid, before was stepwise.

RachelMurray-Watson committed:
Accountde for skipping baseline

RachelMurray-Watson committed:
Changed plot to be scatter plot with error bars

See no significant difference in these plots

RachelMurray-Watson committed:
Can now set what analysis is beig done (main text, scenario SA, parameter SA).

Removed unneccessary duplication of efforts, including unnecessary graphs

Added plots:

1) Showing stacked difference in total deaths and dalys

RachelMurray-Watson committed:
Added saving of % disrupted

RachelMurray-Watson committed:
Added graph of types of treatments affected

RachelMurray-Watson committed:
Does only certain ones

RachelMurray-Watson committed:
Git

RachelMurray-Watson committed:
Changed name of graph

RachelMurray-Watson committed:
For clarity deleted baseline

Also added in other climate params that need to be updated in scenario as they were overwritten in parameter grid

RachelMurray-Watson committed:
Param grid

RachelMurray-Watson committed:
Added in ability to go between climate/parameter/main text

RachelMurray-Watson committed:
Saved breakdown of HSIs in numbers

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:
uses guaranteed disruptions, and guaranteed reseeking healthcare

if year >= 2010:
p_disruption = 1
will_seek_care = 1

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:
uses guaranteed disruptions, and guaranteed reseeking healthcare

if year >= 2010:
p_disruption = 1
will_seek_care = 1

simplified scenario file

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:
uses guaranteed disruptions, and guaranteed reseeking healthcare

if year >= 2010:
p_disruption = 1
will_seek_care = 1

simplified scenario file

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:
uses guaranteed disruptions, and guaranteed reseeking healthcare

if year >= 2010:
p_disruption = 1
will_seek_care = 1

simplified scenario file

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:
uses guaranteed disruptions, and guaranteed reseeking healthcare

if year >= 2010:
p_disruption = 1
will_seek_care = 1

simplified scenario file

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:
uses guaranteed disruptions, and guaranteed reseeking healthcare

if year >= 2010:
p_disruption = 1
will_seek_care = whatever it would have been

simplified scenario file

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:
uses guaranteed disruptions, and guaranteed reseeking healthcare

if year >= 2010:
p_disruption = 0
will_seek_care = 0 (can't happen)

simplified scenario file

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:
uses guaranteed disruptions, and guaranteed reseeking healthcare

if year >= 2010:
p_disruption = 0
will_seek_care = 0 (can't happen)

changed to floats so will run

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:
uses guaranteed disruptions, and guaranteed reseeking healthcare

if year >= 2010:
p_disruption = 1
will_seek_care = whatever it would have been

changed to floats so will run

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:
uses guaranteed disruptions, and guaranteed reseeking healthcare

if year >= 2010:
p_disruption = 1
will_seek_care = 1

changed to floats so will run

RachelMurray-Watson committed:
Introduced variation around scale factor of +- 2 days (but cannot go lower than 1)

RachelMurray-Watson committed:
Now has monthly log instead of yearly

RachelMurray-Watson committed:
ruff format .

ruff check . --fix

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:

P_disruption 0.5
Original Healthcare Seeking

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:

P_disruption = 0.05
Healthcare Seeking = 0.5

RachelMurray-Watson committed:
SENSITIVITY ANALYSIS:

P_disruption = 0.05
Healthcare Seeking = 1

RachelMurray-Watson committed:
For test, set the default p(disruption) and healthcare, but used 2025 disruption

RachelMurray-Watson committed:
SA

RachelMurray-Watson committed:
Introduce a new handling of missed HSIs then rescheduled. Does same check of consumable availability etc

RachelMurray-Watson committed:
Added in dose to make it work

RachelMurray-Watson committed:
Added in dose to make it work

RachelMurray-Watson committed:
Edited the did not run to work

RachelMurray-Watson committed:
Test with no climate disruptions

RachelMurray-Watson committed:
Test with no climate disruptions

RachelMurray-Watson committed:
S

RachelMurray-Watson committed:
Switched back to orginial CMD module after issue was identified

RachelMurray-Watson committed:
Worst case scenario

Use SSP 5.85 high

Set scale factor for seeking to be 60

All other climate parameters set to 1

Add in Mode 2, so some appointments get cancelled. Have naive priority policy, and rescale capabilities.

RachelMurray-Watson committed:
Worst case scenario

Use SSP 5.85 high

Set scale factor for seeking to be 60

All other climate parameters set to 1

Add in Mode 2, so some appointments get cancelled. Have naive priority policy, and rescale capabilities.

RachelMurray-Watson committed:
Worst case scenario

Use SSP 5.85 high

Set scale factor for seeking to be 60

All other climate parameters set to 1

Add in Mode 2, so some appointments get cancelled. Have naive priority policy, and rescale capabilities.

RachelMurray-Watson committed:
RESTORED CORRECT CLIMATE DISRUPTIONS IN HEALTHCARE SYSTEM

RachelMurray-Watson committed:
Added new method, "did_not_run_weather", which writes to the debug logger. It doesn't take people off treatment, and still allows the climate delay to reschedule the event (rescheduling in the did_not_run would, at present, set up two steams of referrals).

RachelMurray-Watson committed:
REMOVED NEW METHOD JUST TO GET RUNS RUNNING

RachelMurray-Watson committed:
Test azure run - no climate effects

RachelMurray-Watson committed:
Added in record of non_blank footprints for full detail HSIs

RachelMurray-Watson committed:
Edited to allow for multiple bars on the disruption graph

RachelMurray-Watson committed:
Edited to allow for multiple bars on the disruption graph

RachelMurray-Watson committed:
Added to allow for climate-specific HSI investigation

RachelMurray-Watson committed:
Re-did parameters for more-memory run

RachelMurray-Watson committed:
Edited to allow for SA

RachelMurray-Watson committed:
Formatting and fixed error where upper bound was not calculated

RachelMurray-Watson committed:
Added formatting in log files for monthly saving of DALYs

RachelMurray-Watson committed:
Add in check for mode_2 SA

RachelMurray-Watson committed:
New colours and names

RachelMurray-Watson committed:
Removed unnecessary extract_results

Added option for mode 2 analysis

RachelMurray-Watson committed:
Included option for mode 2
added in vmin and vmax again for clarity

RachelMurray-Watson committed:
New runs for parameter scan with mode 2

RachelMurray-Watson committed:
Tried to reduce logging burden

RachelMurray-Watson committed:
Fix number_of_draws to use parameter grid length

RachelMurray-Watson committed:
Removed monthly logging to try and ensure all runs

RachelMurray-Watson committed:
Test memory

RachelMurray-Watson committed:
Test memory

RachelMurray-Watson committed:
Test memory

RachelMurray-Watson committed:
Test memory

RachelMurray-Watson committed:
Test memory

RachelMurray-Watson committed:
for mode 2

RachelMurray-Watson committed:
Set up for ssp 5.85

RachelMurray-Watson committed:
Ensured sum of all dalys

RachelMurray-Watson committed:
edited to be neater

RachelMurray-Watson committed:
Changed to DALYs per 1000

RachelMurray-Watson committed:
Fixed up to allow for most recent analyses (baseline, climate ssps, parameter scans, all with mode 2)

RachelMurray-Watson committed:
Fixed up to allow for most recent analyses (baseline, climate ssps, parameter scans, all with mode 2)

RachelMurray-Watson committed:
New parameter analysis fix up

RachelMurray-Watson committed:
No place of residence in recent runs...

RachelMurray-Watson committed:
Now have monthly for mode 2 with baseline

RachelMurray-Watson committed:
Plots HSI by cause with and plots significant differences between ssp and baseline

RachelMurray-Watson committed:
Updated so colours have cervical cancer

RachelMurray-Watson committed:
Add in graph of fine treatment IDs to see which are affected most

RachelMurray-Watson committed:
Refactor and tidy

RachelMurray-Watson committed:
add in potential for monthly dalys

RachelMurray-Watson committed:
Look at total, delayed, and cancelled HSI

RachelMurray-Watson committed:
Save CIs

Check which have significantly different plots

RachelMurray-Watson committed:
Changed date

RachelMurray-Watson committed:
Added python file for ease
